File: neo4j-detection/src/test/java/com/neo4j/data/importer/DockerNeo4j.java
package com.neo4j.data.importer;

public class DockerNeo4j {

    public static String image() {
        return String.format("neo4j:%s", tag());
    }

    private static String tag() {
        String version = version();
        if (enterprise()) {
            return String.format("%s-enterprise", version);
        }
        return version;
    }

    static String version() {
        String version = System.getenv("NEO4J_VERSION");
        if (version == null) {
            return "4.4";
        }
        return version;
    }

    static boolean enterprise() {
        return Boolean.parseBoolean(System.getenv("ENTERPRISE"));
    }
}

File: core/src/test/kotlin/org/neo4j/caniuse/DockerNeo4j.kt
package org.neo4j.caniuse

object DockerNeo4j {
    fun image(): String {
        return String.format("neo4j:%s", tag())
    }

    private fun tag(): String {
        val version = version()
        if (enterprise()) {
            return String.format("%s-enterprise", version)
        }
        return version
    }

    fun version(): String {
        val version = System.getenv("NEO4J_VERSION") ?: return "4.4"
        return version
    }

    fun enterprise(): Boolean {
        return System.getenv("ENTERPRISE").toBoolean()
    }
}
